backend/api/activity/index.js

```diff
@@ -6,68 +6,224 @@ const activity = express.Router();
+	  getobj.startDateTime = req.body.startDateTime || getobj.startDateTime;
+	  getobj.endDateTime = req.body.endDateTime || getobj.endDateTime;
+	  getobj.lastEditDateTime = currentDateTime;
+	  
+	  //Ensure end time comes after start time
+	  if (getobj.endDateTime < getobj.startDateTime) {
+		return res.status(400).json({
+          error: 'endDateTime is less than startDateTime',
+        });
+	  }
+	  
+	  //Save it to the database
+	  await getobj.save();
+	  
+	  //Response
+	  res.status(statusCode).json({
+        activity: getobj,
+		shortened,
+      });
+    } catch (error) {
+	  //Error Response
+      res.status(500).send(error);
+    }
+  });
+  
+/**
+ * Deletes an activity by id
+ * @param {Hex} id - The id of the activity, must be only hexadecimal
+ */
+activity.route('/delete/:id([a-f0-9]+)')
+  .post(async (req, res) => {
+	//Ensure user is authenticated
+	if (!req.user) {
+      return res.status(401).json({
+        error: 'Not authenticated',
+      });
+    }
+	
+	try {
+	  //Get the activity to edit
+      const getobj = await Activity.findById(req.params.id).exec();
+	  
+	  //Ensure the authenticated user is author of this activity
+	  if (getobj.postedBy != req.user.id) {
+		return res.status(403).json({
+          error: 'Wrong account',
+        });
+	  }
+	  
+	  //Delete it
+	  await Activity.deleteOne(getobj);
+	  
+	  //Response
+	  res.sendStatus(200);
+    } catch (error) {
+	  //Error Response
+      res.status(500).send(error);
+    }
   });
 
 /**
- * Searches posts by criteria specified in post data
- * TODO: Add search criteria
- * @body {Number} numPosts - The number of posts to return
+ * Searches activites by custom criteria
+ * @body {Integer} page - The page to send back (Items sent are based on page and pageSize)
+ * @body {Integer} pageSize - The number of activities to return
+ * @body {Integer} sort - The sorting method (see below)
+ * @body {Integer} reverse - Reverses the order of the sort
+ * @body {Integer} omitEnded - Omits activities that have already ended. Defaults to 0.
+ * @body {Integer} omitStarted - Omits activities that have already started. Defaults to 0.
+ *
+ * Sorting methods:
+ * 0 - Start time, soonest first
+ * 1 - Date created, newest first
  */
 activity.route('/search')
   .get(async (req, res) => {
-    const numPosts = req.body.numPosts || 20;
-    try {
-      const getobj = await Activity
-        .find({ creationDateTime: { $exists: true } })
-        .sort({ creationDateTime: -1 })
-        .limit(numPosts);
-      const statusCode = 200;
+	//Default variables for search
+	const page = req.body.page || 1;
+    const pageSize = req.body.pageSize || 20;
+	const sort = req.body.sort || 0;
+	const reverse = req.body.reverse || 0;
+	const omitEnded = req.body.omitEnded || 0;
+	const omitStarted = req.body.omitStarted || 0;
+	const currentDateTime = Date.now();
+	const statusCode = 200;
+	
+	//Handle reversed data
+	var rev = 1;
+	if (reverse) {
+		rev = -1;
+	}
```

Comment:
Why do we have both `rev` and `reverse`? We can force `reverse` to be a boolean value by just checking if it supplied to the request or not

Reply:
rev is not a boolean.

If reverse is 0, rev is 1
If reverse is 1, rev is -1

When picking sort order in mongoose, you can either pass in 1 or -1. With this method, I multiply rev by the default setting to switch the default sort order from 1 to -1 or vice versa.

Reply:
Okay, that's fine then. We can keep it as the values instead